activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -1023,7 +1023,11 @@ def mismatched_foreign_key(message, sql:, binds:, connection_pool:)
         end
 
         def version_string(full_version_string)
-          full_version_string.match(/^(?:5\.5\.5-)?(\d+\.\d+\.\d+)/)[1]
+          if full_version_string && matches = full_version_string.match(/^(?:5\.5\.5-)?(\d+\.\d+\.\d+)/)
+            matches[1]
+          else
+            raise ActiveRecordError, "Unable to parse MySQL version from '#{full_version_string}'"
```

Comment:
Not sure if there is any reason to define a separate class for this exception. I don't see any but commenting in case if someone thinks it deserves a separate error constant

Reply:
Seemed like overkill. I looked for a more specific existing exception that would match but I don't see one. I'm happy to change the class if anyone has a well formed opinion.